packages/xml-views-validation/test/validators/element/non-stable-id-spec.ts

```diff
@@ -0,0 +1,172 @@
+import { partial } from "lodash";
+import { UI5SemanticModel } from "@ui5-language-assistant/semantic-model-types";
+import { generateModel } from "@ui5-language-assistant/test-utils";
+import { generate } from "@ui5-language-assistant/semantic-model";
+import { getMessage, NON_STABLE_ID } from "../../../src/utils/messages";
+import {
+  assertNoIssues as assertNoIssuesBase,
+  assertSingleIssue as assertSingleIssueBase,
+} from "../../test-utils";
+import { validateNonStableId } from "../../../src/validators/elements/non-stable-id";
+
+describe("the use of non stable id validation", () => {
+  let ui5SemanticModel: UI5SemanticModel;
+
+  before(async () => {
+    ui5SemanticModel = await generateModel({
+      version: "1.74.0",
+      modelGenerator: generate,
+    });
+  });
+
+  context("true positive scenarios", () => {
+    let assertSingleIssue: (xmlSnippet: string, message: string) => void;
+    before(() => {
+      assertSingleIssue = partial(
+        assertSingleIssueBase,
+        ui5SemanticModel,
+        {
+          element: [validateNonStableId],
+        },
+        "NonStableIDIssue",
+        "error"
+      );
+    });
+
+    it("will detect missing stable id in non-whitelisted UI5 class", () => {
+      assertSingleIssue(
+        `<mvc:View xmlns:uxap="sap.uxap" xmlns:m="sap.m"
+          xmlns:mvc="sap.ui.core.mvc"
+          xmlns="sap.ui.commons">
+            <🢂m:Panel🢀>
+            </m:Panel>
+        </mvc:View>`,
+        getMessage(NON_STABLE_ID, "Panel")
+      );
+    });
+
+    it("will detect missing stable id in custom control", () => {
+      assertSingleIssue(
+        `<🢂custom:View🢀
```

Comment:
won't it be better to provide a more "real life" example where the custom class
replaces the Button instead of the top level view?

Reply:
this comment does not seem to e resolved

Reply:
we need this test for root-level scenario